internal/sync2/cycle.go

```diff
@@ -124,8 +143,11 @@ func (cycle *Cycle) Run(ctx context.Context, fn func(ctx context.Context) error)
 
 // Close closes all resources associated with it.
 func (cycle *Cycle) Close() {
-	cycle.Stop()
-	<-cycle.stop
+	cycle.initialize()
+	if cycle.runningState.IsRunning() {
+		cycle.Stop()
+		<-cycle.stop
+	}
 	close(cycle.control)
```

Comment:
there is no need to close the channel, because doing so, `Close` panics if it's called more than once.

In master, the cals after the first one, hang forever, but I think that's a bug.

Reply:
I've removed the call to close. And this PR should remove the hanging bug :-)

Reply:
yes, I've realized about that it was going to be fixed if `close` got removed.